alerting/src/main/kotlin/org/opensearch/alerting/resthandler/RestGetSuggestionsAction.kt

```diff
@@ -0,0 +1,97 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.alerting.resthandler
+
+import org.apache.logging.log4j.LogManager
+import org.opensearch.alerting.AlertingPlugin
+import org.opensearch.alerting.action.GetSuggestionsAction
+import org.opensearch.alerting.action.GetSuggestionsRequest
+import org.opensearch.alerting.model.suggestions.suggestioninputs.util.SuggestionInput
+import org.opensearch.alerting.model.suggestions.suggestioninputs.util.SuggestionInputType
+import org.opensearch.client.node.NodeClient
+import org.opensearch.common.xcontent.XContentParser
+import org.opensearch.common.xcontent.XContentParser.Token
+import org.opensearch.common.xcontent.XContentParserUtils.ensureExpectedToken
+import org.opensearch.rest.BaseRestHandler
+import org.opensearch.rest.BaseRestHandler.RestChannelConsumer
+import org.opensearch.rest.RestHandler.Route
+import org.opensearch.rest.RestRequest
+import org.opensearch.rest.RestRequest.Method.POST
+import org.opensearch.rest.action.RestToXContentListener
+
+private val log = LogManager.getLogger(RestGetSuggestionsAction::class.java)
+
+class RestGetSuggestionsAction : BaseRestHandler() {
+
+    override fun getName(): String = "get_suggestions_action"
+
+    override fun routes(): List<Route> {
+        return listOf(
+            Route(POST, AlertingPlugin.SUGGESTIONS_BASE_URI) // inline object with specific format is required
+        )
+    }
+
+    override fun prepareRequest(request: RestRequest, client: NodeClient): RestChannelConsumer {
+        log.debug("${request.method()} ${AlertingPlugin.SUGGESTIONS_BASE_URI}")
+
+        return RestChannelConsumer { channel ->
```

Comment:
I think this whole block below which does the field parsing using `xcp` should be moved into a static `parse` method in `GetSuggestionsRequest` which returns a `GetSuggestionsRequest` object. Then we can just do something like:

```kotlin
val xcp = request.contentParser()
val getSuggestionsRequest = GetSuggestionsRequest.parse(xcp)
```

Reply:
Done, though my static parse accepts a RestRequest rather than a contentParser because part of the parsing requires getting a fresh parser that points to the beginning, so I need to call request.contentParser() twice.